bundles/org.openhab.binding.unifiedremote/src/main/java/org/openhab/binding/unifiedremote/internal/UnifiedRemoteHandler.java

```diff
@@ -0,0 +1,150 @@
+                        break;
+                    case SEND_KEY_CHANNEL:
+                        response = connection.sendKey(stringCommand);
+                        break;
+                    default:
+                        return;
+                }
+                if (isErrorResponse(response)) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Session expired");
+                    connection.authenticate();
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Connection not initialized");
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            if (e.getCause() instanceof ConnectException) {
+                // we assume thing is offline
+                updateStatus(ThingStatus.OFFLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unexpected exception: " + e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        connection = getNewConnection();
+        initConnectionChecker();
+    }
+
+    private UnifiedRemoteConnection getNewConnection() {
+        config = getConfigAs(UnifiedRemoteConfiguration.class);
+        return new UnifiedRemoteConnection(this.httpClient, config.host);
+    }
+
+    private void initConnectionChecker() {
+        stopConnectionChecker();
+        connectionCheckerSchedule = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                ThingStatus status = thing.getStatus();
+                if ((status == ThingStatus.OFFLINE || status == ThingStatus.UNKNOWN) && connection != null) {
+                    connection.authenticate();
+                    updateStatus(ThingStatus.ONLINE);
+                } else if (status == ThingStatus.ONLINE) {
+                    if (isErrorResponse(connection.keepAlive())) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Keep alive failed");
+                    }
+                }
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                if (e.getCause() instanceof ConnectException) {
+                    // we assume thing is offline
+                    updateStatus(ThingStatus.OFFLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Unexpected exception: " + e.getMessage());
+                }
+            }
+        }, 0, 40, TimeUnit.SECONDS);
+    }
+
+    private void stopConnectionChecker() {
+        if (connectionCheckerSchedule != null && !connectionCheckerSchedule.isCancelled()) {
```

Comment:
No harm in canceling a task that is already cancelled.
```suggestion
        if (connectionCheckerSchedule != null) {
```